src/toil_scripts/gatk_processing/gatk_preprocessing.py

```diff
@@ -7,433 +7,239 @@
+    if 'genome.dict' not in config:
+        config['genome.dict'] = job.addChildJobFn(picard_create_sequence_dictionary, fileStoreID, mock=mock).rv()
+    return config
 
 
-def download_sample_gatk(job, shared_ids, input_args, sample):
+def samtools_view(job, bamFileStoreID, flag='0', mock=False):
     """
-    Defines sample variables then downloads the sample.
+    Outputs bam file using a samtools view flag value
+    :param job: Job instance
+    :param bamFileStoreID str: bam fileStoreID
+    :param flag str: samtools defined flags
+    :param mock bool: If True, run in mock mode
+    :return str: bam fileStoreID
 
-    ids: dict           Dictionary of fileStore IDs
-    input_args: dict    Dictionary of input arguments
-    sample: str         Contains uuid, url
+    '0x800'
     """
-    uuid, url = sample
-    # Create a unique
-    input_args.uuid = uuid
-    if not input_args.output_dir.startswith('s3://'):
-        input_args.output_dir = os.path.join(input_args.output_dir, uuid)
-    # Download sample bams and launch pipeline
-    if input_args.ssec:
-        shared_ids['sample.bam'] = job.addChildJobFn(download_encrypted_file, url, 'sample.bam', input_args.ssec).rv()
-    else:
-        shared_ids['sample.bam'] = job.addChildJobFn(download_from_url_gatk, url=url, name='sample.bam').rv()
-    job.addFollowOnJobFn(remove_supplementary_alignments, shared_ids, input_args)
-
-def remove_supplementary_alignments(job, shared_ids, input_args):
     work_dir = job.fileStore.getLocalTempDir()
-
-    #Retrieve file path
-    read_from_filestore(job, work_dir, shared_ids, 'sample.bam')
-    outpath = os.path.join(work_dir, 'sample.nosuppl.bam')
+    inputs = {'sample.bam': bamFileStoreID}
+    outputs = {'sample.output.bam': None},
+    get_files_from_filestore(job, work_dir, inputs)
+    outpath = os.path.join(work_dir, 'sample.output.bam')
 
     command = ['view',
-               '-b', '-o', '/data/sample.nosuppl.bam',
-               '-F', '0x800',
+               '-b', '-o', '/data/sample.output.bam',
+               '-F', flag,
                '/data/sample.bam']
-
     docker_call(work_dir=work_dir, parameters=command,
                 tool='quay.io/ucsc_cgl/samtools:1.3--256539928ea162949d8a65ca5c79a72ef557ce7c',
-                inputs=['sample.bam'],
-                outputs={'sample.nosuppl.bam': None})
-    shared_ids['sample.nosuppl.bam'] = job.fileStore.writeGlobalFile(outpath)
-    job.addChildJobFn(sort_sample, shared_ids, input_args)
-
+                inputs=inputs.keys(),
+                outputs=outputs,
+                mock=mock)
+    return job.fileStore.writeGlobalFile(outpath)
 
-def _move_bai(outpath):
 
-    # bam index gets written without bam extension
-    os.rename(outpath.replace('bam', 'bai'), outpath + '.bai')
-
-
-def sort_sample(job, shared_ids, input_args):
+def picard_sort_sam(job, fileStoreID, config, xmx=8):
```

Comment:
Move to `toil_scripts.tools.preprocessing`.
Also, I don't think we need this with the way we're running alignment? Should come back and check this.